docs/manual/src/docs/asciidoc/_includes/servlet/saml2/saml2-login.adoc

```diff
@@ -223,6 +223,66 @@ public interface Saml2AuthenticationRequestFactory {
 }
 ----
 
+=== Customizing Authentication Logic
+
+By default Spring Security configures the `OpenSamlAuthenticationProvider`
+to validate and parse the SAML 2 response and assertions that are received.
+This provider has three customization options
+
+1. An authorities extractor - extract group information from the assertion
+2. An authorities mapper - map extracted group information to internal authorities
+3. Response time validation duration - the built in tolerances for timestamp validation
+should be used when there may be a time synchronization issue.
+
+The simplest customization is to use an `ObjectPostProcessor`
+
+[source,java]
+----
+@EnableWebSecurity
+public class SecurityConfig extends WebSecurityConfigurerAdapter {
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        ObjectPostProcessor<OpenSamlAuthenticationProvider> processor = new ObjectPostProcessor<>() {
+            @Override
+            public <O extends OpenSamlAuthenticationProvider> O postProcess(O provider) {
+                provider.setResponseTimeValidationSkew(RESPONSE_TIME_VALIDATION_SKEW);
+                provider.setAuthoritiesMapper(AUTHORITIES_MAPPER);
+                provider.setAuthoritiesExtractor(AUTHORITIES_EXTRACTOR);
+                return provider;
+            }
+        };
+
+        http.saml2Login()
+                .addObjectPostProcessor(processor)
+        ;
+        super.configure(http);
+    }
+}
+----
+
+The authentication manager for the security filter can also be overwritten, using your own
```

Comment:
This sentence seems to imply by omission that the user's only alternative is creating a custom `AuthenticationManager` instance.

It seems to me that the following is perfectly acceptable and arguably as easy as the first option:

```java
OpenSamlAuthenticationProvider authenticationProvider = new OpenSamlAuthenticationProvider(...);
authenticationProvider.setResponseTimeValidationSkew(...)
...
AuthenticationManager authenticationManager = new ProviderManager
    (Arrays.asList(authenticationProvider));
http
    .saml2Login()
        .authenticationManager(authenticationManager);
```

Reply:
It's almost the same. The default filter chain will have a `ProviderManager` with more than one authentication manager. This replaces the provider manager with a singleton list.

I like this suggestion and will add it to the documentation